util.sh

```diff
@@ -13,6 +13,16 @@ fi
 
 SOURCE=${BASH_SOURCE:-}
 
+if [[ $(echo $BASH_VERSION | head -c1) -lt 5 ]]; then
```

Comment:
why just warn and not exit?

Reply:
not every command might fail so thought to be graceful about it but its prolly better to exit so pushed that